src/TextModelProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext } from 'react';
+
+export interface ITextModel {
+  getSource(): string;
+  setSource(source: string): void;
+}
+
+type TextModelState = {
+  model?: ITextModel;
+};
+
+const TextModelContext = createContext<TextModelState | undefined>(undefined);
```

Comment:
Does this mean that task lists work in the markdown preview as well?

Reply:
Yes & no. RenderMime isn't supposed to write back to the model IIRC, but we could probably do it if the model we're given is directly associated with the file content, i.e. not a copy. If this were the case, we would need to do a bit more here, though.